csrc/scheduler/matmul_utils.cpp

```diff
@@ -578,48 +580,32 @@ std::shared_ptr<MatmulParams> getMatmulHeuristics(
   // Check initial conditions
   const auto fusion_exprs = fusion->exprs();
   auto mma_exprs = ir_utils::filterByType<MmaOp>(fusion_exprs).vector();
-  if (mma_exprs.size() != 1) {
-    // Support only for fusion with a single mma op
-    return nullptr;
-  }
+  TORCH_INTERNAL_ASSERT(
+      mma_exprs.size() == 1, "Support only fusion with a single mma op.");
 
   const auto layout = getInputsLayout(mma_exprs.front());
-  if (layout.second) {
-    // Layout check returned an error message
-    if (isDebugDumpEnabled(DebugDumpOption::MatmulChecks)) {
-      printMsg(layout.second.value());
-    }
-    return nullptr;
-  }
+  TORCH_INTERNAL_ASSERT(!layout.second.has_value(), layout.second.value());
 
   const auto problem_shape = getProblemShape(
       fusion, mma_exprs[0]->as<MmaOp>(), runtime_info, layout.first.value());
-  if (!problem_shape) {
-    // Failed to acquire problem shape
-    return nullptr;
-  }
+  TORCH_INTERNAL_ASSERT(
+      problem_shape.has_value(), "Failed to acquire problem shape.");
 
   const auto device_prop = at::cuda::getCurrentDeviceProperties();
   const auto mma_op = getMmaOp(
       device_prop->major * 10 + device_prop->minor, problem_shape.value());
-  if (!mma_op) {
-    // No heuristics can be prepared if mma op request is empty
-    return nullptr;
-  }
+  TORCH_INTERNAL_ASSERT(
```

Comment:
Fail early is more convenient for debugging.

Reply:
Ok, it makes sense - I wasn't sure about the desired execution flow.

Reply:
PyTorch usually uses two macros to check for unexpected result, `TORCH_CHECK` and `TORCH_INTERNAL_ASSERT `. A `TORCH_CHECK` usually indicates that there is a user error, a `TORCH_INTERNAL_ASSERT` usually means there is a bug in PyTorch. For our case here, we should use `TORCH_INTERNAL_ASSERT` because if these functions can not get the information they want, the fusion should have been rejected, and `getMatmulHeuristics` shouldn't have been called at all.